### PR TITLE
chore(engines): drop Node 14 support

### PR DESCRIPTION
**Pull request by tanner-reits**
## What is the current behavior?

Stencil currently support Node 14

GitHub Issue Number: N/A

## What is the new behavior?

Minimum version Stencil supports is now Node 16 🎉  

## Does this introduce a breaking change?

- [X] Yes
- [ ] No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -125,8 +125,8 @@
     "ws": "8.13.0"
   },
   "engines": {
-    "node": ">=14.10.0",
-    "npm": ">=6.0.0"
+    "node": ">=16.0.0",
+    "npm": ">=7.10.0"
```
Comment (tanner-reits) [inline package.json:129]: Pulled these from Node's [previous releases matrix](https://nodejs.org/en/download/releases)

**Review by tanner-reits — COMMENTED** [review]
Just noting a few things. Need CI to run to make sure things are looking good and need to test the `fs` changes still. 

Also, the docs site was already updated for v4 node support so don't need to make any changes over there

**Review by alicewriteswrongs — APPROVED** [review]

File: test/karma/package.json [hunk 2]
```diff
@@ -45,7 +45,7 @@
     "workbox-build": "4.3.1"
   },
   "volta": {
-    "node": "14.21.1",
-    "npm": "6.14.18"
+    "node": "16.20.0",
+    "npm": "8.19.4"
```
Comment (rwaskiewicz) [inline test/karma/package.json:49]: Can we bump `@types/node` while we're in here? ATM, we're pinned at v14

Comment (rwaskiewicz) [reply test/karma/package.json:49]: In doing this (bump) I believe that'll update the `package-lock.json` to v2 of the package lock format. If that's the case, can you bump the `package-lock.json` files of under `test/karma` too? I think it's just:
- `test/karma/test-sibling`
- `test/karma/test-invisible-prehydration`

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]
Can you do me a favor and fill out the testing steps section of this PR for me please?

**Review by rwaskiewicz — APPROVED** [review]

**Pull request merged**